packages/atomic/src/components/result-template-components/atomic-result-value/atomic-result-value.tsx

```diff
@@ -12,17 +12,38 @@ import {
 export class AtomicResultValue {
   @Element() host!: HTMLDivElement;
   @Prop() value = '';
+  @Prop() shouldHighlightWith = '';
 
   @ResultContext() private result!: Result;
 
+  /**
+   * Returns the value of 'shouldHighlightWith' prop from the Result
+   */
+  private getHighlightsFromResult() {
+    const highlightsKeywords = Object.entries(this.result).find(
+      (keyValue) => keyValue[0] === this.shouldHighlightWith
+    );
+    return highlightsKeywords![1] as string[];
+  }
```

Comment:
Ideally, this would not exist and be replaced by `this.result[this.shouldHighlightWith]` on line 42. But I gave me an error.( `type 'string' can't be used to index type 'Result'`). So, if you guys know a trick that I don't for the issue, lmk 😄

Reply:
I'd move that whole method to headless.
There you can create a new type to limit options: 

```typescript
export type ResultHighlights =
  | 'titleHighlights'
  | 'firstSentencesHighlights'
  | 'excerptHighlights'
  | 'printableUriHighlights'
  | 'summaryHighlights';

export function getHighlightsFromResult(
  result: Pick<Result, ResultHighlights>,
  highlight: ResultHighlights
) {
  const highlightsKeywords = Object.entries(result).find(
    (keyValue) => keyValue[0] === highlight
  );
  return highlightsKeywords![1];
}

```

Adding validation for the `shouldHighlightWith` prop, and type it to `ResultHighlights` is also a +!

Reply:
I'm blind lol. There's already a function to qet a property from the result `ResultTemplatesHelpers.getResultProperty()` used already in the render() of `atomic-result-value` 🤦  No need for a seconde one 😅